src/servers/src/http/handler.rs

```diff
@@ -27,14 +27,26 @@ use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 use session::context::QueryContextRef;
 
-use crate::http::{ApiState, GreptimeOptionsConfigState, JsonResponse};
+use crate::http::{ApiState, Epoch, GreptimeOptionsConfigState, JsonResponse, ResponseFormat};
 use crate::metrics_handler::MetricsHandler;
 use crate::query_handler::sql::ServerSqlQueryHandlerRef;
 
 #[derive(Debug, Default, Serialize, Deserialize, JsonSchema)]
 pub struct SqlQuery {
     pub db: Option<String>,
     pub sql: Option<String>,
+    // (Optional) result format: [`gerptimedb_v1`, `influxdb_v1`],
+    // the default value is `greptimedb_v1`
+    pub format: Option<String>,
```

Comment:
Ref - https://github.com/GreptimeTeam/greptimedb/issues/577

I wonder if this format can be extended to `csv` also? Perhaps we can discuss a result pattern like the payload contains bare text and move metadata to the HEADERs.

Reply:
cc @MichaelScofield

Reply:
If we go further complex in this greptimedb_v1 and influxdb_v1 way, we assume heavily that the result is in JSON format.

Reply:
The format of csv is fixed, right? We cannot have csv formats with different outputs for the same table data.

Reply:
Yeah. My question is that whether these two "format" the same concept.

It seems that we can still have a "format=csv" and lower the current `sql` function as an `inner_sql` then adapt format if needed.

Reply:
It looks to me "format=csv" can be added just like the other formats, and the results from sql execution can be formated like others. Why add an "inner_sql" function?

Reply:
The current `fn sql` returns `Json<JsonResponse>`, but "format=CSV" would return other than JSON and also custom Headers that needs an extra layer.

It's an implementation details, though. We can flatten the result of `JsonResponse` and call into response in `fn sql` instead. Will a dedicate function would be smooth for testings.

Reply:
Oh, I see. Then we do need to refactor the "sql" function to adapt both json and csv. I think for csv format, we can simply return http plain text, can we?

Reply:
Yes. Thanks for your information. My initial intention is to be clear that this PR doesn't affect #577 we can still return results in CSV format.

Then,

- format unset or `=greptimedb_v1`, the current manner
- format `=influxdb_v1`, the manner introduced in this PR
- format `=csv`, output follows greptimedb_v1, but in a format that payload is plain text with meta in headers.

Reply:
LGTM, cc @fengjiachun

Reply:
Also LGTM